### ISSUE TITLE
Lint/NextWithoutAccumulator is reported incorrectly for a nested loop

### ISSUE THREAD

**Issue by leonid-shevtsov** [bbatsov/rubocop#3308]
## Expected behavior

Having a `reduce` loop and another loop inside of it, it's normal to `next` without an argument from the nested loop. It's only wrong to `next` without an argument from the `reduce` loop.

## Actual behavior

`Lint/NextWithoutAccumulator` is reported.

## Steps to reproduce the problem

Example code to reproduce:

```
result = [[1, 2, 3], [4, 5, 6]].inject([]) do |acc, elems|
  elems.each_with_index do |elem, i|
    next if i == 1
    acc << elem
  end
  acc
end

puts result
```

gets us

```
Inspecting 1 file
W

Offenses:

repro.rb:3:5: W: Lint/NextWithoutAccumulator: Use next with an accumulator argument in a reduce.
    next if i == 1
    ^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

```
$ rubocop -V
0.37.2 (using Parser 2.3.0.6, running on ruby 2.1.2 x86_64-darwin14.0)```
